Examples.md

```diff
@@ -21,6 +21,20 @@ int main(void) {
 }
 ```
 
+## C++
+
+```c++
+#include <cstdlib>
+#include <iostream>
+
+int main(int, char**)
+{
+  std::cout << "Hello World!\n" << std::flush;
+
+  return (std::cout.rdstate()) ? EXIT_FAILURE : EXIT_SUCCESS;
```

Comment:
Can I ask why you used `.rdstate()` and converted it to bool, instead of converting `std::cout` to bool, or any of the other ways of testing for failure? I myself am unable to understand the differences between all the ways, so it would help me to have an explanation that I can point to if anyone ever asks me a question about this :-}.

Reply:
Thanks!